tests/support/proxy.rs

```diff
@@ -62,6 +72,26 @@ impl Proxy {
         self
     }
 
+    pub fn dns<R2>(self, dns: R2) -> Proxy<R2>
```

Comment:
Are we able to do something with implementing `default()` for a `Resolver` so that we don't need separate types for what should be the same thing?

Reply:
Hmm, if I'm understanding what you mean correctly, I don't think so --- we need to have a `NewResolver` trait rather than requiring a bound like `trait Resove: Default` because some resolvers (i.e. the actual dns resolver) also return a background future from their constructors.